src/ucp/proto/proto_common.c

```diff
@@ -646,12 +646,16 @@ ucp_lane_index_t ucp_proto_common_find_lanes_with_min_frag(
 }
 
 void ucp_proto_common_add_proto(const ucp_proto_common_init_params_t *params,
-                                const ucp_proto_caps_t *proto_caps,
-                                const void *priv, size_t priv_size)
+                                ucp_proto_perf_t *perf, const void *priv,
+                                size_t priv_size)
 {
-    ucp_proto_select_add_proto(&params->super, params->cfg_thresh,
-                               params->cfg_priority, proto_caps, priv,
-                               priv_size);
+    ucs_status_t status = ucp_proto_select_add_proto(&params->super,
+                                                     params->cfg_thresh,
+                                                     params->cfg_priority,
+                                                     perf, priv, priv_size);
+    if (status != UCS_OK) {
+        ucp_proto_perf_destroy(perf);
```

Comment:
Please call destroy on the same level as create.

Reply:
Please elaborate, do you suggest to return status from `ucp_proto_common_add_proto` and destroy perf on one level higher if `status != OK` ?

Reply:
Done

Reply:
Yes, exactly.
`ucp_proto_common_add_proto` is just a wrapper of `ucp_proto_select_add_proto`. It is better to change the signature of the function to return the status.
The current implementation will work only because `ucp_proto_common_add_proto` is called in the end of a caller function. But it may lead to an error, when the caller function will try to use `perf` after it was destroyed inside `ucp_proto_common_add_proto`.

Reply:
btw, do we really need this wrapper then?

Reply:
Removed